### ISSUE TITLE
Pull choices value from other fields in the survey

### ISSUE THREAD

**Issue by B-Cheye** [onaio/onadata#1899]
### Environmental Information

Ona.io production and MoDa production

### Problem description
I have been working on a form that we're filtering options from the choices sheet based on the number of entered names, however, I have a challenge when I apply a function in the choice-filter column. The choices are filtered but the names are not pulled as expected. Instead of the entered name, we are getting the reference eg ${hhmem1}.Kindly assist in further troubleshooting the issue.
### Expected behavior
The choices should be able to be pulled values entered and not the function.
### Steps to reproduce the behavior
You can replicate this by uploading this form on either Ona or MoDa and making some submissions 
[choice_filter.xlsx](https://github.com/onaio/onadata/files/5237405/choice_filter.xlsx)

### Additional Information
Note that the same function works well on Kobo. 

_Logs, [related issues](github.com/onaio/onadata/issues), weird / out of place occurrences, local settings, possible approach to solving this..._

**Comment by DavisRayM** [comment]
https://github.com/XLSForm/pyxform/issues/403

**Comment by alepietrobon** [comment]
hey, is this getting assigned, resourced, looked into? @DavisRayM

**Comment by DavisRayM** [comment]
Yes, it's currently being looked into @alepietrobon. We're currently discussing possible approaches to solving this.

_[Slack conversation thread](https://onaio.slack.com/archives/C4S0D2S3H/p1601290935020200)_

**Comment by alepietrobon** [comment]
Awesome. Thanks, Davis. Let me know here or in the thread once you have a
proposed solution.
Downgrading does feel like a workaround but we're already experiencing this
issue with more than one client so we might want to confirm if there are
alternatives.Thanks!

**Comment by DavisRayM** [comment]
This isn't supported in the XLSForm spec. A new question field was added into pyxform recently to support usage of a previous questions answers as choices.

**Comment by msschroeder** [comment]
@DavisRayM Can you share the link to this - for us to review. Also, if this will not be fixed - Doesn't that mean we would want to close the issue?
